clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -256,11 +256,13 @@ public void poll(long timeout, long now, PollCondition pollCondition) {
      * nor will it execute any delayed tasks.
      */
     public void pollNoWakeup() {
-        disableWakeups();
-        try {
-            poll(0, time.milliseconds(), null);
-        } finally {
-            enableWakeups();
+        synchronized (this) {
```

Comment:
@hachikuji I think the `close` action on the `ConsumerCoordinator` will become somewhat tricky to adjust now that I tried this out in practice. It calls `maybeAutoCommitOffsetsSync` which calls `commitOffsetsSync` which triggers a `poll` as well as `ensureCoordinatorReady` which also triggers a `poll` potentially. So we would have to hand that parameter down to through those methods as well wouldn't we?